packages/contentstack-bulk-publish/src/producer/publish-entries.js

```diff
@@ -37,14 +42,31 @@ async function getEntries(stack, contentType, locale, bulkPublish, environments,
       .then(async (entriesResponse) => {
         skipCount += entriesResponse.items.length;
         let entries = entriesResponse.items;
-        for (let index = 0; index < entriesResponse.items.length; index++) {
+
+        for (let index = 0; index < entries.length; index++) {
+          let variants = [];
+
+          if (variantsFlag) {
+            const variantsEntriesResponse = await stack
+              .contentType(contentType)
+              .entry(entries[index].uid)
+              .variants()
+              .query(queryParams)
+              .find();
```

Comment:
@sunil-lakshman , then & catch block is missing.  Please create a separate function and manage pagination as well, rather than adding variation code to the same block.
if(variantsFlag){
   variants = await getVariantEntries();
}
async function getVariantEntries(){
// handle then & catch
}